src/c1_koiter_steigmann/koiter_steigmann_curvable_bell_elements.h

```diff
@@ -9,580 +9,1183 @@
+            if (fabs(coeff) > Orthogonality_tolerance)
+            {
+              // [zdec] debug
+              std::cout << "eqn " << condition_index
+			<< " depends on dof L" << left_duid2_index
+                        << std::endl;
+              left_data_used[left_duid2_index] = true;
+              continue;
+            }
+          }
+          if (!left_duid1d1_pinned && !left_data_used[left_duid1d1_index])
+          {
+            double coef =
+              jac_of_transform(0, alpha) * jac_of_transform(0, beta);
+            if (fabs(coef) > Orthogonality_tolerance)
+            {
+              // [zdec] debug
+              std::cout << "eqn " << condition_index
+			<< " depends on dof L" << left_duid1d1_index
+                        << std::endl;
+              left_data_used[left_duid1d1_index] = true;
+              continue;
+            }
+          }
+          if (!left_duid1d2_pinned && !left_data_used[left_duid1d2_index])
+          {
+            double coef =
+              jac_of_transform(0, alpha) * jac_of_transform(1, beta) +
+              jac_of_transform(1, alpha) * jac_of_transform(0, beta);
+            if (fabs(coef) > Orthogonality_tolerance)
+            {
+              // [zdec] debug
+              std::cout << "eqn " << condition_index
+			<< " depends on dof L" << left_duid1d2_index
+                        << std::endl;
+              left_data_used[left_duid1d2_index] = true;
+              continue;
+            }
+          }
+          if (!left_duid2d2_pinned && !left_data_used[left_duid2d2_index])
+          {
+            double coef =
+              jac_of_transform(1, alpha) * jac_of_transform(1, beta);
+            if (fabs(coef) > Orthogonality_tolerance)
+            {
+              // [zdec] debug
+              std::cout << "eqn " << condition_index
+			<< " depends on dof L" << left_duid2d2_index
+                        << std::endl;
+              left_data_used[left_duid2d2_index] = true;
+              continue;
+            }
+          }
+          // -------------------------------------------------------------------
+          // if we made it here, it is because all dofs in the constraint are
+          // pinned so we need to check the constraint is satisfied manually
+          // and then remove it by pinning the corresponding lagrange
+          // multiplier
 
-    /// Constructor: Call constructors for C1CurvedBellElement and
-    /// KoiterSteigmannEquations
-    KoiterSteigmannC1CurvableBellElement()
-      : CurvableBellElement<Nnode_1D>(Nfield, Field_is_bell_interpolated),
-        KoiterSteigmannEquations()
-    {
-      // // Use the higher order integration scheme
-      // delete this->integral_pt();
-      // // Do we want something that is order 8 instead?
-      // TGauss<2, 9>* new_integral_pt = new TGauss<2, 9>;
-      // this->set_integration_scheme(new_integral_pt);
+          // hierher Aidan should this stay alive? 
```

Comment:
Answered earlier

Reply:
Aidan: please tidy up brutally; I'll keep this comment in there to remind you but will mark this as resolved